src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -289,7 +289,7 @@ public function searchAction(Request $request): Response
             return new RedirectResponse($this->generator->generate('_profiler', ['token' => $token]), 302, ['Content-Type' => 'text/html']);
         }
 
-        $tokens = $this->profiler->find($ip, $url, $limit, $method, $start, $end, $statusCode);
+        $tokens = $this->profiler->find($ip, $url, (int) $limit, $method, $start, $end, $statusCode);
```

Comment:
changes on this file are not needed IMHO

Reply:
Why not? This variable now got from the request, so it should be a string, am I right?
https://github.com/symfony/symfony/blob/ca088d47cb4662335dde91914a82497091a2279d/src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php#L239

Reply:
because type casts are weaker than auto-casts performed by php so they're a dangerous practice
this would cast null to 0, which we do not want

Reply:
So looks like the validation should go before calling `find` method and after getting the payload from the request in that case. But afaiu, this is the WebProfiler bundle, for me, it looks safe enough for that type of cast. I am wrong?

Reply:
Anyway, this is unrelated the linked issue so I'm not sure why we're discussing this here.

Reply:
Okay, reverted these changes and rebased them.